taichi/rhi/cuda/cuda_device.cpp

```diff
@@ -76,13 +76,14 @@ void CudaDevice::dealloc_memory(DeviceAllocation handle) {
   }
 }
 
-void *CudaDevice::map(DeviceAllocation alloc) {
+RhiResults CudaDevice::map(DeviceAllocation alloc, void *&mapped_ptr) {
   AllocInfo &info = allocations_[alloc.alloc_id];
   size_t size = info.size;
   info.mapped = new char[size];
   // FIXME: there should be a better way to do this...
   CUDADriver::get_instance().memcpy_device_to_host(info.mapped, info.ptr, size);
-  return info.mapped;
+  mapped_ptr = info.mapped;
+  return RhiResults::success;
```

Comment:
nit: do we always return success here?

Reply:
Actual return status has not been implemented on anything other than Vulkan atm. Don't want to make this MR larger than it is already